File: src/main/java/fi/aalto/cs/apluscourses/presentation/FileSaveViewModel.java
package fi.aalto.cs.apluscourses.presentation;

import com.intellij.openapi.vfs.VirtualFile;
import java.nio.file.Path;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class FileSaveViewModel {

  @NotNull
  private final String title;

  @NotNull
  private final String description;

  @Nullable
  private final VirtualFile defaultDirectory;

  @Nullable
  private final String defaultName;

  private Path path;

  /**
   * Construct a view model with the given parameters.
   * @param defaultDirectory An optional default directory.
   * @param defaultName      An optional default name for the file.
   */
  public FileSaveViewModel(@NotNull String title,
                           @NotNull String description,
                           @Nullable VirtualFile defaultDirectory,
                           @Nullable String defaultName) {
    this.title = title;
    this.description = description;
    this.defaultDirectory = defaultDirectory;
    this.defaultName = defaultName;
  }

  @NotNull
  public String getTitle() {
    return title;
  }

  @NotNull
  public String getDescription() {
    return description;
  }

  @Nullable
  public VirtualFile getDefaultDirectory() {
    return defaultDirectory;
  }

  @Nullable
  public String getDefaultName() {
    return defaultName;
  }

  @NotNull
  public Path getPath() {
    return path;
  }

  public void setPath(@NotNull Path path) {
    this.path = path;
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/ui/temp/presentation/FileSaveViewModel.kt
package fi.aalto.cs.apluscourses.ui.temp.presentation

import com.intellij.openapi.vfs.VirtualFile
import java.nio.file.Path

class FileSaveViewModel(
    title: String,
    description: String,
    defaultDirectory: VirtualFile?,
    defaultName: String?
) {
    private val title: String

    private val description: String

    private val defaultDirectory: VirtualFile?

    private val defaultName: String?

    private var path: Path? = null

    /**
     * Construct a view model with the given parameters.
     *
     * @param defaultDirectory An optional default directory.
     * @param defaultName      An optional default name for the file.
     */
    init {
        this.title = title
        this.description = description
        this.defaultDirectory = defaultDirectory
        this.defaultName = defaultName
    }

    fun getTitle(): String {
        return title
    }

    fun getDescription(): String {
        return description
    }

    fun getDefaultDirectory(): VirtualFile? {
        return defaultDirectory
    }

    fun getDefaultName(): String? {
        return defaultName
    }

    fun getPath(): Path {
        return path!!
    }

    fun setPath(path: Path) {
        this.path = path
    }
}
